README.md

```diff
@@ -20,104 +38,162 @@ for the following use cases:
+```
+results=$(sonobuoy retrieve)
+```
 
-* Your $PATH configured:
+Remove the Sonobuoy pod and wait for the namespace to be removed:
+```
+sonobuoy delete --wait
+```
 
+Inspect results for test failures.  This will list the number of tests failed and their names:
+```
+sonobuoy e2e $results
 ```
-$ export PATH=$GOROOT/bin:$GOPATH/bin:$PATH 
-```  
 
-[gimme]: https://github.com/travis-ci/gimme
+### Checking Cluster Conformance
+
+To run Sonobuoy to check if your cluster passes Kubernetes conformance, run:
+```
+sonobuoy run
+```
+
+This is the command which will generate the data necessary to validate your
+cluster is conformant according to the [CNCF][cncf].
 
-### Download and run
+### Quick Mode
 
-Download the CLI by running:
+Often you don't want to wait for an entire conformance run to complete but want
+some minimal check that the cluster is behaving and you have the proper
+configurations in place to run conformance in the future.
 
+In that case, run:
 ```
-$ go get -u -v github.com/heptio/sonobuoy
+sonobuoy run --mode quick
 ```
 
-Deploy a Sonobuoy pod to your cluster with:
+This runs only the e2e plugin (Kubernetes tests) but specifies that it should
+only run a single, short test. This should normally take less than a minute to complete.
 
+### Monitoring Sonobuoy during a run
+You can check on the status of each of the plugins running with:
 ```
-$ sonobuoy run
+sonobuoy status
 ```
 
-View actively running pods:
+You can also inspect the logs of all Sonobuoy containers:
+```
+sonobuoy logs
+```
 
+### Getting Results
+To view the output, copy the output directory from the aggregator Sonobuoy pod to
+your local machine (and save the name of the file to a variable for reference):
 ```
-$ sonobuoy status 
+output=$(sonobuoy retrieve)
 ```
 
-To inspect the logs:
+The results can be inspected without being extracted. It will list the number of tests failed and their names:
+```
+sonobuoy e2e $output
+```
+
+You can also extract the output locally so that you can view the other
+information Sonobuoy gathered as well:
+ - detailed plugin results
+ - pod logs
+ - query results about the contents/state of your cluster
 
 ```
-$ sonobuoy logs
+mkdir ./results; tar xzf $output -C ./results
 ```
 
-To view the output, copy the output directory from the main Sonobuoy pod to
-somewhere local:
+For information on the contents of the snapshot, see the [snapshot
+documentation][snapshot].
+
+### Cleanup
+
+Sonobuoy creates a few resources in order to run and expects to run within its
+own namespace.
 
+Cleaning up Sonobuoy entails removing it's namespace as well as a few cluster
+scoped resources.
+
+Delete the main namespace and cluster resources by running:
 ```
-$ sonobuoy retrieve .
+sonobuoy delete 
 ```
 
-This copies a single `.tar.gz` snapshot from the Sonobuoy pod into your local
-`.` directory. Extract the contents into `./results` with:
+There are some Kubernetes e2e tests that may leak some resources. Sonobuoy can
```

Comment:
Move this to `docs` section. Also, needs to make sure to call out how it does delete with `--all` (i.e. it deletes any namespace starting with `e2e*`)

Reply:
Moved to a known issue section and shortened.